src/www/app/outline_server_repository/access_key_serialization.ts

```diff
@@ -31,6 +31,58 @@ export function staticKeyToShadowsocksSessionConfig(staticKey: string): Shadowso
       password: config.password.data,
     };
   } catch (error) {
-    throw new errors.ServerUrlInvalid(error.message || 'Failed to parse static access key.');
+    throw new errors.ServerAccessKeyInvalid(error.message || 'Failed to parse static access key.');
+  }
+}
+
+function maybeParseShadowsocksSessionConfigJson(maybeJsonText: string): ShadowsocksSessionConfig | null {
```

Comment:
```suggestion
function parseShadowsocksSessionConfigJson(configJson: string): ShadowsocksSessionConfig | null {
```

Drop the "maybe". It forces me to look in the implementation to see when you are or are not parsing. But you are always parsing. You may return null if invalid, but that doesn't warrant the maybe name.